alwaysontop/main.js

```diff
@@ -38,27 +56,55 @@ function onAlwaysOnTopWindow(
         const win = event.newGuest = new BrowserWindow(
             Object.assign(options, {
                 backgroundColor: 'transparent',
-                width: SIZE.width,
-                height: SIZE.height,
                 minWidth: SIZE.width,
                 minHeight: SIZE.height,
-                maxWidth: SIZE.width,
-                maxHeight: SIZE.height,
                 minimizable: false,
                 maximizable: false,
-                resizable: false,
+                resizable: true,
                 alwaysOnTop: true,
                 fullscreen: false,
                 fullscreenable: false,
                 skipTaskbar: true,
                 titleBarStyle: undefined,
                 frame: false,
                 show: false
-            }, getPosition())
+            }, getPosition(), getSize())
         );
         win.once('ready-to-show', () => {
             win.showInactive();
         });
+
+        //for macOS we use the built-in setAspectRatio on resize, for other we use custom implementation
+        if (os.type() === 'Darwin') {
+            win.setAspectRatio(16/9);
+        } else {
+            win.on('will-resize', (e, newBounds) => {
+                oldSize = win.getSize();
+                const mousePos = robot.getMousePos();
+                const windowBottomRightPos = {
+                    x: newBounds.x + newBounds.width - 16,
+                    y: newBounds.y + newBounds.height - 16,
+                };
+                //prevent resize from bottom right corner as it is buggy.
+                if (mousePos.x >= windowBottomRightPos.x && mousePos.y >= windowBottomRightPos.y) {
```

Comment:
I'm wondering isn't it there a better way to detect this? 

I'm worried that you can end up in the same situation even if you started the resize from another corner. Although probably we shouldn't care about this.

Reply:
Not sure if it s possible to end up in this situation from another corner, as the check is done on future position of the window,  and the future position doesn't seem possible to have the bottom right corner 'upper' and 'lefter' from current mouse position :)